src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { props, t } from '../utils';
+import { Input, InputRequiredProps, InputDefaultProps, Props as InputPropsT } from '../Input/Input';
+import View from 'react-flexview';
+
+export type PasswordInputRequiredProps = InputRequiredProps;
+
+export type PasswordInputDefaultProps = InputDefaultProps & {
+  hideText: string,
+  showText: string
+};
+
+export namespace PasswordInput {
+  export type Props = PasswordInputRequiredProps & Partial<PasswordInputDefaultProps>;
+}
+type PasswordInputDefaultedProps = PasswordInputRequiredProps & PasswordInputDefaultProps;
+
+export const Props = {
+  ...InputPropsT,
+  hideText: t.maybe(t.String),
+  showText: t.maybe(t.String)
+};
+
+export type State = {
+  show: boolean
+}
+
+@props(Props, { strict: false })
+export class PasswordInput extends React.PureComponent<PasswordInput.Props, State> {
+
+  static defaultProps: PasswordInputDefaultProps = {
+    disabled: false,
+    showText: 'Show',
+    hideText: 'Hide'
+  };
+
+  state = {
+    show: false
+  };
+
+  onToggleClick = () => {
+    this.setState(({ show }) => ({ show: !show }));
+  }
+
+  render() {
+    const props = this.props as PasswordInputDefaultedProps;
+    const type = this.state.show ? 'text' : 'password';
+    const text = this.state.show ? props.hideText : props.showText;
+    const toggleProps = {
+      onClick: this.onToggleClick,
+      className: 'toggle',
```

Comment:
this `className` is named as one of our components => `Toggle`'s CSS properties may be applied to this.
CSS modules may be a long-term solution for these problems, but for now I'd simply rename it